### PR TITLE
chore: Closed date banners

### PR DESCRIPTION
**Pull request by timarney**
# Summary | Résumé

- Adds admin banner to note the date a form was closed

Screenshot 2024-10-17 at 9 09 36 AM

- Adds front facing banner to denote when a scheduled form closing date will happen. 

screenshot_2024-10-17_at_11 57 30___am_720

☝️ 

Note to test you'll need to directly modify a closing date in your database

### CODE REVIEW

File: i18n/translations/fr/form-builder.json [hunk 1]
```diff
@@ -884,6 +884,9 @@
       "errors": {
         "translation": "Veuillez fournir du contenu équivalent dans les deux langues officielles."
       }
+    },
+    "banner": {
```
Comment (timarney) [inline i18n/translations/fr/form-builder.json:888]: @anikbrazeau some new line here

Again hoping you can swap around these placeholders for FR

```
"text2": "{{month}} {{day}}, {{year}} at {{hour}}:{{minute}} EST. [FR",
```

**Review by thiessenp-cds — DISMISSED** [review]
Nice! Especially the Utils will be helpful. Some tests may help but I think those can be done in a future PR. LGTM :)

File: i18n/translations/en/common.json [hunk 2]
```diff
@@ -196,5 +196,11 @@
     "type2": "NEW",
     "message2": "Build complex multi-page forms with display logic and branching. <a href='https://articles.alpha.canada.ca/forms-formulaires/setting-up-form-logic/' className='text-white hover:text-white'>Learn more</a>"
   },
+  "closingNotice": {
+    "title": "Form Closing",
+    "text1": "Form will be taking submissions up to",
+    "text2": "{{month}} {{day}}, {{year}} at {{hour}}:{{minute}} EST.",
+    "text3": "After this date, the form is closed."
+  },
```
Comment (anikbrazeau) [inline i18n/translations/en/common.json:204]:
```suggestion
    "title": "Form closing",
    "text1": "Form accepting submissions until:",
    "text2": "{{month}} {{day}}, {{year}} at {{hour}}:{{minute}} ET.",
    "text3": "After this date, the form is closed."
  },
```

File: i18n/translations/en/common.json [hunk 3]
```diff
@@ -196,5 +196,11 @@
     "type2": "NEW",
     "message2": "Build complex multi-page forms with display logic and branching. <a href='https://articles.alpha.canada.ca/forms-formulaires/setting-up-form-logic/' className='text-white hover:text-white'>Learn more</a>"
   },
+  "closingNotice": {
+    "title": "Form Closing",
+    "text1": "Form will be taking submissions up to",
+    "text2": "{{month}} {{day}}, {{year}} at {{hour}}:{{minute}} EST.",
+    "text3": "After this date, the form is closed."
```
Comment (anikbrazeau) [inline i18n/translations/en/common.json:203]: Do we need the string "text3"?

File: i18n/translations/en/form-builder.json [hunk 4]
```diff
@@ -884,6 +884,9 @@
       "errors": {
         "translation": "Provide equivalent content in both official languages."
       }
+    },
+    "banner": {
+      "text": "Form closed on {{month}} {{day}}, {{year}} at {{hour}}:{{minute}} EST"
```
Comment (anikbrazeau) [inline i18n/translations/en/form-builder.json:889]:
```suggestion
    },
    "banner": {
      "text": "Form closed on {{month}} {{day}}, {{year}} at {{hour}}:{{minute}} ET"
```

File: i18n/translations/fr/common.json [hunk 5]
```diff
@@ -196,5 +196,11 @@
     "type2": "NOUVEAUTÉ",
     "message2": "Créez des formulaires complexes de plusieurs pages avec logique d’affichage et d’embranchement. <a href='https://articles.alpha.canada.ca/forms-formulaires/fr/configurer-formulaire-avec-logique/' className='text-white hover:text-white'>En savoir plus</a>"
   },
+  "closingNotice": {
+    "title": "Form Closing [FR]",
+    "text1": "Form will be taking submissions up to [FR",
+    "text2": "{{month}} {{day}}, {{year}} at {{hour}}:{{minute}} EST. [FR",
+    "text3": "After this date, the form is closed. [FR"
+  },
```
Comment (anikbrazeau) [inline i18n/translations/fr/common.json:204]:
```suggestion
    "title": "Fermeture du formulaire",
    "text1": "Les soumissions sont acceptées jusqu'au :",
    "text2": "{{day}} {{month}} {{year}} à {{hour}} h {{minute}} HE",
    "text3": "Passé cette date, le formulaire est fermé."
  },
```

File: i18n/translations/fr/form-builder.json [hunk 6]
```diff
@@ -884,6 +884,9 @@
...
+      "text": "Form closed on {{month}} {{day}}, {{year}} at {{hour}}:{{minute}} EST [FR]"
```
Comment (anikbrazeau) [inline i18n/translations/fr/form-builder.json:889]:
```suggestion
      "text": "Le formulaire a été fermé le {{day}} {{month}} {{year}} à {{hour}} h {{minute}} HE"
```

File: i18n/translations/en/common.json (hunk 3 above)
Comment (timarney) [reply i18n/translations/en/common.json]: removed - as discussed not needed.

**Pull request merged**